### PR TITLE
Host preflight

### PR DESCRIPTION
**Pull request by areed**

### CODE REVIEW

File: examples/preflight/host-disk-usage.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+              message: /var/lib/kubelet has less than 10Gi of disk space available
+          - fail:
+              when: "used/total > 70%"
+              message: /var/lib/kubelet is more than 70% full
+          - warn:
+              when: "total < 40Gi"
+              message: /var/lib/kubelet has less than 40Gi of total space
+          - warn:
+              when: "used/total > 60%"
+              message: /var/lib/kubelet is more than 50% full
```
Comment (laverya) [inline examples/preflight/host-disk-usage.yaml]: 60 and 50, pick one

File: examples/preflight/sample-host-preflight.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        timeout: 5000ms
+    - diskUsage:
+        collectorName: ephemeral
+        path: /var/lib/kubelet
+  analyzers:
+    - cpu:
+        outcomes:
+          - fail:
+              when: "< 4"
+              message: This server has less than 4 CPU cores, and we require 8, but recommend 16
```
Comment (laverya) [inline examples/preflight/sample-host-preflight.yaml:20]: I feel like this example used to have another option

File: pkg/analyze/host_cpu.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+				return &result, nil
+			}
+		}
+	}
+
+	return &result, nil
+}
+
+func compareHostCPUConditionalToActual(conditional string, logicalCount int, physicalCount int) (res bool, err error) {
```
Comment (laverya) [inline pkg/analyze/host_cpu.go:102]: Could we get a unit test for this? There's one for doCompareHostCPU, which is a lot simpler

**Review by laverya — APPROVED** [review]

File: examples/preflight/sample-host-preflight.yaml (hunk 2 above)
Comment (areed) [reply examples/preflight/sample-host-preflight.yaml:20]: I added examples/preflight/host-cpu.yaml with more examples

File: pkg/analyze/host_cpu.go (hunk 3 above)
Comment (areed) [reply pkg/analyze/host_cpu.go:102]: Added

**Pull request merged**